internal/impl/mail/input_pop3.go

```diff
@@ -0,0 +1,299 @@
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, countParam, err)
+	}
+	interval, err := conf.Namespace(pullParam).FieldDuration(intervalParam)
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, intervalParam, err)
+	}
+	maxInFlight, err := conf.FieldInt(maxInFlightParam)
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, maxInFlightParam, err)
+	}
+	timeout, err := conf.FieldDuration(timeoutParam)
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, timeoutParam, err)
+	}
+	maxBodySize, err := conf.FieldInt(maxBodySizeParam)
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, maxBodySizeParam, err)
+	}
+	listPageSize, err := conf.FieldInt(listPageSizeParam)
+	if err != nil {
+		return nil, fmt.Errorf(errorPattern, listPageSizeParam, err)
+	}
+
+	client := pop3.New(pop3.Opt{
+		Host:          host,
+		Port:          port,
+		TLSEnabled:    tlsEnabled,
+		TLSSkipVerify: tlsSkipCertVerify,
+		DialTimeout:   timeout,
+	})
+
+	p := &pop{
+		client:       client,
+		username:     username,
+		password:     password,
+		maxBodySize:  maxBodySize,
+		listPageSize: listPageSize,
+		count:        count,
+		interval:     interval,
+		messageChan:  make(chan *envelope, maxInFlight),
+		done:         make(chan struct{}),
+		ack:          make(chan int, maxInFlight),
+		log:          log,
+	}
+
+	return p, nil
+}
+
+// Connect attempts to establish a connection and fetch the emails periodically
+func (p *pop) Connect(ctx context.Context) error {
+	p.onlyOnce.Do(func() {
+		go func() {
+			defer func() {
+				p.done <- struct{}{}
+			}()
+			ticker := time.NewTicker(p.interval)
+			times := 0
+			for ; ; <-ticker.C {
+				for hasMessages := true; hasMessages; {
+					n, err := p.fetchEmails()
+					if err != nil {
+						p.log.Errorf("error fetching emails: %v", err)
+					}
+					hasMessages = n > 0 && n >= p.listPageSize
+				}
+				times++
+				if p.count > 0 && times >= p.count {
+					ticker.Stop()
+					break
+				}
+			}
+		}()
+	})
+	return nil
+}
+
+func (p *pop) fetchEmails() (int, error) {
+	c, err := p.client.NewConn()
+	if err != nil {
+		return 0, fmt.Errorf("error establishing the connection: %v", err)
+	}
+	defer func() {
+		if err := c.Quit(); err != nil {
+			p.log.Errorf("error quiting: %v", err)
+		}
+	}()
+	if err := c.Auth(p.username, p.password); err != nil {
+		return 0, fmt.Errorf("error while logging in: %v", err)
+	}
+	msgs, err := c.List(0)
+	if err != nil {
+		return 0, fmt.Errorf("error listing emails: %v", err)
+	}
+	var wg sync.WaitGroup
+	for _, m := range msgs {
+		entity, err := c.Retr(m.ID)
+		if err != nil {
+			p.log.Errorf("error retrieving email %d: %v", m.ID, err)
+		}
+		body := make([]byte, p.maxBodySize)
+		n, err := entity.Body.Read(body)
+		if err != nil {
+			p.log.Debugf("error reading email body %d: %v", m.ID, err)
+		}
+		message := service.NewMessage(body[:n])
+		for k, v := range entity.Header.Map() {
+			message.MetaSet(k, v[0])
```

Comment:
Rather than use the keys exactly as they are I'm considering whether we should match the style of keys from other inputs, so it'd be `pop3_return_path` rather than `Return-Path`.